### PR TITLE
feat: language filter

### PR DESCRIPTION
**Pull request by GeronimoJohn**
# Description

### <samp>🤖 Generated by Copilot at c75af0a</samp>

This pull request adds a new feature to the journeys-admin app that allows the user to filter the template gallery by language. It also adds a new option to hide the main panel header for the template gallery page. It implements the UI components, the logic, and the tests for the new feature, and modifies the existing components to integrate the feature. It affects the following files: `PageWrapper.tsx`, `PageWrapper.spec.tsx`, `TemplateSections.tsx`, `templates/index.tsx`, `TemplateGallery.tsx`, and the files in the `LanguageFilter` and `LanguageFilterDialog` folders.

[- Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/34356579/todos/6557661695)

# How should this PR be QA Tested?

- [ ] it should enable users to click on language button
- [ ] it should enable users to filter by language via the language filter dialog
- [ ] it should render templates according to the set language filter
- [ ] it should show empty placeholder component when there's no template relating to language filter

# Walkthrough

### <samp>🤖 Generated by Copilot at c75af0a</samp>

*  Add a new feature to allow hiding the main panel header in the `PageWrapper` component ([link](https://github.com/JesusFilm/core/pull/1965/files?diff=unified&w=0#diff-4d089d7265256280ff64e3b201d5e5b93de7ec08609a68128848f06ced5a0effR25), [link](https://github.com/JesusFilm/core/pull/1965/files?diff=unified&w=0#diff-2ce7fd8777e1be5b5ec657940e0b2e29ae6b6fb4d8d88f12d4142e8bb0f2aa3dR38-R48), [link](https://github.com/JesusFilm/core/pull/1965/files?diff=unified&w=0#diff-5965e5f2f61464ac798b2d978eb50c82874afd08713ad90b1fbd61c962f04237R22), [link](https://github.com/JesusFilm/core/pull/1965/files?diff=unified&w=0#diff-5965e5f2f61464ac798b2d978eb50c82874afd08713ad90b1fbd61c962f04237R41), [link](https://github.com/JesusFilm/core/pull/1965/files?diff=unified&w=0#diff-5965e5f2f61464ac798b2d978eb50c82874afd08713ad90b1fbd61c962f04237L99-R109))
* Add a new feature to allow filtering the template gallery by language using the `LanguageFilter` and `LanguageFilterDialog` components ([link](https://github.com/JesusFilm/core/pull/1965/files?diff=unified&w=0#diff-65f857ab32a512ba56278e090018b40fbbe5fbccdfd8d993981d30d25e7f684eR1-R80), [link](https://github.com/JesusFilm/core/pull/1965/files?diff=unified&w=0#diff-13372d81b3e76e196f12415e5cd0054ddb13daeb2cef1392e047609e4ac89a92R1-R80), [link](https://github.com/JesusFilm/core/pull/1965/files?diff=unified&w=0#diff-eade1215e915a226c3d290c1dea9067bde85b34a3044ac37e872858aea0f4e3bR1-R88), [link](https://github.com/JesusFilm/core/pull/1965/files?diff=unified&w=0#diff-292f7e49cefda504f2deda92476407deb657d8baf3a1663e2aae42debc4afd33R1-R117), [link](https://github.com/JesusFilm/core/pull/1965/files?diff=unified&w=0#diff-75b1f19f628a103d6dd0fdc4643290bf0273802a2182891917c26ef28d396043R1), [link](https://github.com/JesusFilm/core/pull/1965/files?diff=unified&w=0#diff-62fcfb9c89b7ecf047c8ad5dec882e8ed46b7264e996fce8deb02da3f3ef6454R1), [link](https://github.com/JesusFilm/core/pull/1965/files?diff=unified&w=0#diff-b3ab650ee0dc0ee335e53cf2f72cf2700dbc0890a665f9b7a7b608c6ce920650L1-R38), [link](https://github.com/JesusFilm/core/pull/1965/files?diff=unified&w=0#diff-8528e7b4c06335c3a765fccde78d68fff718a8b69d7384bea6782deeb45efb8aL20-R31), [link](https://github.com/JesusFilm/core/pull/1965/files?diff=unified&w=0#diff-8528e7b4c06335c3a765fccde78d68fff718a8b69d7384bea6782deeb45efb8aL37-R63), [link](https://github.com/JesusFilm/core/pull/1965/files?diff=unified&w=0#diff-8528e7b4c06335c3a765fccde78d68fff718a8b69d7384bea6782deeb45efb8aL74-R115))
* Modify the `TemplateSections` component to filter the journeys by the `languageId` prop and render the `TemplateSection` component conditionally based on the language and tag filters ([link](https://github.com/JesusFilm/core/pull/1965/files?diff=unified&w=0#diff-8528e7b4c06335c3a765fccde78d68fff718a8b69d7384bea6782deeb45efb8aL37-R63), [link](https://github.com/JesusFilm/core/pull/1965/files?diff=unified&w=0#diff-8528e7b4c06335c3a765fccde78d68fff718a8b69d7384bea6782deeb45efb8aL74-R115))
* Modify the `TemplateGallery` component to include the `LanguageFilter` component in the header and pass the `languageId` query parameter to the `TemplateSections` component ([link](https://github.com/JesusFilm/core/pull/1965/files?diff=unified&w=0#diff-b3ab650ee0dc0ee335e53cf2f72cf2700dbc0890a665f9b7a7b608c6ce920650L1-R38))
* Add test cases for the `PageWrapper`, `LanguageFilter`, and `LanguageFilterDialog` components ([link](https://github.com/JesusFilm/core/pull/1965/files?diff=unified&w=0#diff-2ce7fd8777e1be5b5ec657940e0b2e29ae6b6fb4d8d88f12d4142e8bb0f2aa3dR38-R48), [link](https://github.com/JesusFilm/core/pull/1965/files?diff=unified&w=0#diff-65f857ab32a512ba56278e090018b40fbbe5fbccdfd8d993981d30d25e7f684eR1-R80), [link](https://github.com/JesusFilm/core/pull/1965/files?diff=unified&w=0#diff-eade1215e915a226c3d290c1dea9067bde85b34a3044ac37e872858aea0f4e3bR1-R88))

### CODE REVIEW

**Review by tataihono — CHANGES_REQUESTED** [review]

File: apps/journeys-admin/src/components/TemplateGallery/LanguageFilter/LanguageFilter.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import { ReactElement, useState } from 'react'
+
+import AddSquare4Icon from '@core/shared/ui/icons/AddSquare4'
+
+import { GetLanguages } from '../../../../__generated__/GetLanguages'
+import { GET_LANGUAGES } from '../../Editor/EditToolbar/Menu/LanguageMenuItem/LanguageDialog'
+
+import type { LanguageFilterDialogProps } from './LanguageFilterDialog'
+
+const DynamicLanguageFilterDialog = dynamic<LanguageFilterDialogProps>(
```
[the remark this answers is not in the document]
Comment (GeronimoJohn) [reply apps/journeys-admin/src/components/TemplateGallery/LanguageFilter/LanguageFilter.tsx:14]: I was trying to get the language api call to happen when the component is needed. But my implementation seems to be wrong
Screenshot 2023-10-18 at 2 52 03 PM

[the remark this answers is not in the document]
Comment (GeronimoJohn) [reply apps/journeys-admin/src/components/TemplateGallery/LanguageFilter/LanguageFilter.tsx:14]: I just realised what I did haha. The button should be dynamic instead

File: apps/journeys-admin/src/components/NewPageWrapper/PageWrapper.spec.tsx [hunk 2]
```diff
@@ -35,6 +35,17 @@ describe('PageWrapper', () => {
       expect(getByRole('link')).toHaveAttribute('href', '/')
     })
 
+    it('should show main panel header', () => {
+      const { getByText } = render(
+        <MockedProvider>
+          <FlagsProvider>
+            <PageWrapper title="Page Title" showPanelHeader />
```
Comment (tataihono) [inline apps/journeys-admin/src/components/NewPageWrapper/PageWrapper.spec.tsx]:
```suggestion
            <PageWrapper title="Page Title" />
```

File: apps/journeys-admin/src/components/TemplateGallery/TemplateGallery.tsx [hunk 3]
```diff
@@ -8,11 +10,13 @@ import { useTranslation } from 'react-i18next'
 
 import { TemplateSections } from '../TemplateSections'
 
+import { LanguageFilter } from './LanguageFilter'
 import { TagsFilter } from './TagsFilter'
 
 export function TemplateGallery(): ReactElement {
   const { t } = useTranslation('apps-journeys-admin')
   const router = useRouter()
+  const [languageId, setLanguageId] = useState('529')
```
Comment (tataihono) [inline apps/journeys-admin/src/components/TemplateGallery/TemplateGallery.tsx:19]:
```suggestion
  const ENGLISH_LANGUAGE_ID = '529'
  const [languageIds, setLanguageIds] = useState([ENGLISH_LANGUAGE_ID])
```

File: apps/journeys-admin/src/components/TemplateGallery/TemplateGallery.tsx [hunk 4]
```diff
@@ -66,6 +89,7 @@ export function TemplateGallery(): ReactElement {
       </Grid>
       <TemplateSections
         tagIds={selectedTagIds.length > 0 ? selectedTagIds : undefined}
+        languageId={languageId}
```
Comment (tataihono) [inline apps/journeys-admin/src/components/TemplateGallery/TemplateGallery.tsx:92]:
```suggestion
        languageIds={languageIds}
```

File: apps/journeys-admin/src/components/TemplateGallery/TemplateGallery.tsx [hunk 5]
```diff
@@ -32,6 +36,25 @@ export function TemplateGallery(): ReactElement {
...
+      >
+        <Typography variant="h2" sx={{ display: { xs: 'none', lg: 'block' } }}>
+          {t('Journey Templates')}
+        </Typography>
+        <Typography variant="h2" sx={{ display: { xs: 'block', lg: 'none' } }}>
+          {t('Templates')}
+        </Typography>
+        <LanguageFilter
+          languageId={languageId}
+          onChange={(value) => setLanguageId(value)}
```
Comment (tataihono) [inline apps/journeys-admin/src/components/TemplateGallery/TemplateGallery.tsx:55]:
```suggestion
          languageId={languageIds[0]}
          onChange={(value) => setLanguageIds([value])}
```

File: apps/journeys-admin/src/components/TemplateSections/TemplateSections.tsx [hunk 6]
```diff
@@ -17,20 +17,24 @@ interface Contents {
 
 interface TemplateSectionsProps {
   tagIds?: string[]
+  languageId: string
```
Comment (tataihono) [inline apps/journeys-admin/src/components/TemplateSections/TemplateSections.tsx:20]:
```suggestion
  languageIds?: string[]
```

File: apps/journeys-admin/src/components/TemplateSections/TemplateSections.tsx [hunk 7]
```diff
@@ -17,20 +17,24 @@ interface Contents {
...
 }
 
 export function TemplateSections({
-  tagIds
+  tagIds,
+  languageId
```
Comment (tataihono) [inline apps/journeys-admin/src/components/TemplateSections/TemplateSections.tsx:25]:
```suggestion
  languageIds
```

File: apps/journeys-admin/src/components/TemplateSections/TemplateSections.tsx [hunk 8]
```diff
@@ -17,20 +17,24 @@ interface Contents {
...
   const [collection, setCollection] = useState<Journey[]>([])
+
   const { data, loading } = useJourneysQuery({
     variables: {
       where: {
         template: true,
         orderByRecent: true,
-        tagIds
+        tagIds,
+        languageIds: [languageId]
```
Comment (tataihono) [inline apps/journeys-admin/src/components/TemplateSections/TemplateSections.tsx:37]:
```suggestion
        languageIds
```

File: apps/journeys-admin/src/libs/useTagsQuery/index.ts [hunk 9]
```diff
@@ -1 +1 @@
-export { useTagsQuery } from './useTagsQuery'
+export { useTagsQuery, transformTagsIntoTreeTags } from './useTagsQuery'
```
Comment (tataihono) [inline apps/journeys-admin/src/libs/useTagsQuery/index.ts:1]:
```suggestion
export { useTagsQuery } from './useTagsQuery'
```

**Review by tataihono — CHANGES_REQUESTED** [review]

File: apps/journeys-admin/src/components/TemplateGallery/LanguageFilter/LanguageFilterDialog/LanguageFilterDialog.tsx [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+      onSubmit={handleSubmit}
+    >
+      {({ values, handleSubmit, resetForm, setFieldValue }) => (
+        <Dialog
+          open={open}
+          onClose={handleClose(resetForm)}
+          dialogTitle={{ title: 'Edit Language' }}
+          dialogAction={{
+            onSubmit: handleSubmit,
+            closeLabel: 'Cancel'
```
Comment (tataihono) [inline apps/journeys-admin/src/components/TemplateGallery/LanguageFilter/LanguageFilterDialog/LanguageFilterDialog.tsx:77]:
```suggestion
            closeLabel: t('Cancel')
```

File: apps/journeys-admin/src/components/TemplateGallery/LanguageFilter/LanguageFilterDialog/LanguageFilterDialog.tsx [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+      initialValues={{
+        language: languages != null ? getLanguage(languageId) : undefined
+      }}
+      onSubmit={handleSubmit}
+    >
+      {({ values, handleSubmit, resetForm, setFieldValue }) => (
+        <Dialog
+          open={open}
+          onClose={handleClose(resetForm)}
+          dialogTitle={{ title: 'Edit Language' }}
```
Comment (tataihono) [inline apps/journeys-admin/src/components/TemplateGallery/LanguageFilter/LanguageFilterDialog/LanguageFilterDialog.tsx:74]:
```suggestion
          dialogTitle={{ title: t('Filter By Language') }}
```

**Review by tataihono — CHANGES_REQUESTED** [review]

**Review by tataihono — DISMISSED** [review]

**Review by tataihono — DISMISSED** [review]

**Review by tataihono — APPROVED** [review]

**Pull request merged**